server/aws-lsp-codewhisperer/src/shared/utils.ts

```diff
@@ -503,65 +502,100 @@ export function hasConnectionExpired(error: any) {
 /**
   Lists files in a directory respecting gitignore and npmignore rules.
   @param directory The absolute path of root directory.
+  @param logging Optional logging instance for error reporting.
   @returns A promise that resolves to an array of absolute file paths.
  */
-export async function listFilesWithGitignore(directory: string): Promise<string[]> {
-    let ignorePatterns: string[] = [...COMMON_GITIGNORE_PATTERNS]
-
-    // Process .gitignore
-    const gitignorePath = path.join(directory, '.gitignore')
+export async function listFilesWithGitignore(directory: string, logging?: Logging): Promise<string[]> {
     try {
-        const gitignoreContent = await fs.readFile(gitignorePath, { encoding: 'utf8' })
-        ignorePatterns = ignorePatterns.concat(
-            gitignoreContent
-                .split('\n')
-                .map(line => line.trim())
-                .filter(line => line && !line.startsWith('#'))
-        )
-    } catch (err: any) {
-        if (err.code !== 'ENOENT') {
-            console.log('Preindexing walk: gitIgnore file could not be read', err)
-        }
-    }
+        const MAX_FILES = 500_000
+
+        async function findIgnoreFiles(dir: string): Promise<string[]> {
+            const ignoreFiles: string[] = []
+
+            async function walkDir(currentDir: string): Promise<void> {
+                try {
+                    const entries = await fs.readdir(currentDir, { withFileTypes: true })
+
+                    for (const entry of entries) {
+                        const fullPath = path.join(currentDir, entry.name)
+
+                        if (entry.isFile() && (entry.name === '.gitignore' || entry.name === '.npmignore')) {
```

Comment:
we need to add files with common ignore patterns and all the contents in the gitignore and npmignore files